### PR TITLE
Update Microsoft.ContainerService managedClusters API version to 2024-02-01

### PR DESCRIPTION
**Pull request by olofmattsson-inriver**
## Describe your changes

Support for Azure Kubernetes 2023-01-02-preview API is ending 20 June 2024. Update to the latest stable version.

### CODE REVIEW

**Review by zioproto — APPROVED** [review]

**Comment by zioproto** [comment]
This is a valid issue

Screenshot 2024-05-29 at 13 59 13

according to:
https://learn.microsoft.com/en-us/azure/templates/microsoft.containerservice/allversions

the latest API version is correctly 2024-02-01

@lonegunmanb could we please merge this asap ? thanks

File: main.tf [hunk 1]
```diff
@@ -660,7 +660,7 @@ resource "null_resource" "http_proxy_config_no_proxy_keeper" {
 resource "azapi_update_resource" "aks_cluster_http_proxy_config_no_proxy" {
   count = can(var.http_proxy_config.no_proxy[0]) ? 1 : 0
 
-  type = "Microsoft.ContainerService/managedClusters@2023-01-02-preview"
+  type = "Microsoft.ContainerService/managedClusters@2024-02-01-preview"
```
Comment (lonegunmanb) [inline main.tf:663]: You've assigned two different API for two aks update resources, do we have any special reason for this?

**Review by lonegunmanb — COMMENTED** [review]
Thanks @olofmattsson-inriver for opening this pr to us! I have a question for the API version you assigned.

**Comment by lonegunmanb** [comment]
Hi @olofmattsson-inriver the e2e test failed, but I checked the error message, I believe we have fixed the issue which caused error in our latest main branch. Would you please rebase your branch with our latest main branch and try again? Thanks!

**Comment by lonegunmanb** [comment]
I'm going to merge this pr despite e2e test failure, since after a review I'm sure that the failure was caused by our fragile tests.

**Pull request merged**